### DISCUSSION TITLE
How to set the attribute Win64 of RegistrySearch to yes or no according to the bit of Windows Operating System?

### DISCUSSION THREAD

**Discussion by yanpeng** [wixtoolset/issues#6827] [Questions] [answered]
Hi, 

A new WiX user here. 

I use burn to install prerequisites, including the .net framework and vsto runtime. I have to know if I can install these package, so I do some registry search, for example, on 32 bit Windows, the registry of VSTO Runtime is located at `HKLM\SOFTWARE\Microsoft\VSTO Runtime Setup`, but on 64 bit Windows, `HKLM\SOFTWARE\WOW6432Node\Microsoft\VSTO Runtime Setup` instead. 

In this case, the attribute `Win64` of the element `RegistrySearch` should be set to `no` on 32-bit Windows, `yes` for 64-bit Windows.
```
<util:RegistrySearch 
      Id="VSTORuntime" 
      Root="HKLM" 
      Key="SOFTWARE\Microsoft\VSTO Runtime Setup\v4R\" 
      Value="VSTORFeature_CLR40" 
      Variable="VSTORFeature" 
      Win64="xxxxx" />
```

Is there a way to set the attribute `Win64` of `RegistrySearch` according to the bit of OS? btw, I know the build-in `VersionNT64`, but idk how to convert that to `yes` or `no`.

**Comment by robmen** [comment, marked as answer]
I'd query for both and use the two values (one or both of which may be empty) as appropriate.

**Reply by yanpeng** [reply]
Thanks for reply, your idea inspired me. Now I also query them like you, with the `Condition` being set.
```
<util:RegistrySearch 
      Id="VSTORuntime32VersionV4R" 
      Root="HKLM" 
      Key="SOFTWARE\Microsoft\VSTO Runtime Setup\v4R\" 
      Value="Version" 
      Variable="VSTOR32VersionV4R" 
      Condition="NOT VersionNT64" />

    <util:RegistrySearch 
      Id="VSTORuntime64VersionV4R" 
      Root="HKLM" 
      Key="SOFTWARE\Wow6432Node\Microsoft\VSTO Runtime Setup\v4R\" 
      Value="Version" 
      Variable="VSTOR64VersionV4R" 
      Condition="VersionNT64" />
```
